src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -107,11 +110,23 @@ export function mapCadlTypeToCSharpInputTypeKind(
 
 function getCSharpInputTypeKindByIntrinsicModelName(
     name: string,
-    format?: string
+    format?: string,
+    encode?: EncodeData
 ): InputTypeKind {
     switch (name) {
         case "bytes":
-            return InputTypeKind.Bytes;
+            switch (encode?.encoding) {
+                case undefined:
+                case "base64":
+                    return InputTypeKind.Bytes;
+                case "base64":
+                    return InputTypeKind.BytesBase64Url;
+                default:
+                    logger.warn(
```

Comment:
Shall we throw error if the `encode` is not valid?

Reply:
No. if encode is not valid, we just give out warning message, and will return default encode.